pkg/channel/event_receiver.go

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2018 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package channel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/eventing/pkg/utils"
+)
+
+var (
+	shutdownTimeout = 1 * time.Minute
+
+	// ErrUnknownChannel is returned when an event is received by a channel dispatcher for a
+	// channel that does not exist.
+	ErrUnknownChannel = errors.New("unknown channel")
```

Comment:
Let's turn this into a real type. 

```golang
type UnknownChannelError struct {
  c ChannelReferenc
}

func (u *UnknownChannelError) Error() string {
  return fmt.Sprintf("unknown channel: %v", u.c)
}
```

Then below we can have:

```golang
err = r.receiverFunc(sctx, channel, event)
if err != nil {
	if _, ok := err.(UnknownChannel); ok {
		resp.Status = http.StatusNotFound
	} else {
		resp.Status = http.StatusInternalServerError
	}
	return err
}
```

Reply:
Done, thanks!